lib/fluent/plugin/out_amplitude.rb

```diff
@@ -98,6 +100,25 @@ def verify_user_and_device(amplitude_hash)
       present?(user_id) || present?(device_id)
     end
 
+    def set_time!(amplitude_hash, record)
+      return unless @time_key && !@time_key.empty?
+      @time_key.each do |time_key|
+        next unless record[time_key]
+        if (time = parse_time_from_string(record[time_key]))
+          amplitude_hash[:time] = time
+          break
+        end
+      end
+    end
+
+    def parse_time_from_string(time_string)
+      # this should be seconds since epoch; amplitude-api
+      # converts it to milliseconds since epoch as needed.
+      Time.parse(time_string).to_i
```

Comment:
I don't know if this needs to be more complicated? I think it will work for all of our general data, but it might be worth adding some fancier time string parsing here (esp since this is a public plugin)